core/src/main/java/hivemall/fm/FactorizationMachineUDTF.java

```diff
@@ -563,6 +580,10 @@ protected void runTrainingIteration(int iterations) throws HiveException {
                 inputBuf.flip();
 
                 for (int iter = 2; iter <= iterations; iter++) {
+                    if (earlyStopValidation) {
```

Comment:
better to avoid many `if (earlyStopValidation) {`.

`_validateState` can always be non-null when `if(earlyStopValidation && _validateState.isLossIncreased()` never be true.

Reply:
`_validateState = NULL` allows us to confirm if at least one validation sample exists && early stopping option is enabled. Thanks to this implementation, we can avoid unnecessary call of `cacheCurrentModel` which consumes extra memory to hold the best model parameters.